### PR TITLE
chore: Use named circuitbreakers in persistence

### PR DESCRIPTION
**Pull request by johanandren**
References https://github.com/lightbend/cinnamon/issues/1793

### CODE REVIEW

File: akka-persistence/src/main/scala/akka/persistence/Persistence.scala [hunk 1]
```diff
@@ -344,6 +344,14 @@ class Persistence(val system: ExtendedActorSystem) extends Extension {
       case None       => throw new IllegalArgumentException(s"Unknown plugin actor $journalPluginActor")
     }
 
+  private[akka] final def extensionIdFor(journalPluginActor: ActorRef): String =
+    pluginExtensionId.get().collectFirst {
+      case (id, ext) if ext(system).actor == journalPluginActor => id
+    } match {
+      case Some(conf) => conf
+      case None       => throw new IllegalArgumentException(s"Unknown plugin actor $journalPluginActor")
+    }
```
Comment (octonato) [inline akka-persistence/src/main/scala/akka/persistence/Persistence.scala:353]: `getOrElse` would be more straightforward as we are not doing anything in the `Some` case, just unwrapping it.

**Review by octonato — APPROVED** [review]
LGTM

With a style suggestion. Not blocking for merging.

File: akka-persistence/src/main/scala/akka/persistence/Persistence.scala (hunk 1 above)
Comment (johanandren) [reply akka-persistence/src/main/scala/akka/persistence/Persistence.scala:353]: Yeah, I agree, not hot path in any way, mostly followed how the other similar methods are done. Should add internal api markers though.

**Review by pvlugter — APPROVED** [review]
LGTM

**Pull request merged**